### ISSUE TITLE
LaTeX writer does not respect numfig_format option in captions, and links are short

### ISSUE THREAD

**Issue by Jellby** [sphinx-doc/sphinx#1719]
The captions for figures, tables and code-blocks do not change with numfig_format with the LaTeX writer, and even with the default, code-blocks are called "List" instead of "Listing".

For simple formats ("My fig %s", "My tab %s", "My code %s"), it would be as simple as adding to sphinx.sty:

```
\addto\captionsenglish{\renewcommand{\figurename}{My fig}}
\addto\captionsenglish{\renewcommand{\tablename}{My tab}}
\floatname{literal-block}{My code}
```

In addition, the links generated by LaTeX span only the number, and not the whole name+number pair, as with the HTML writer. This could be fixed by writing:

```
\hyperref[label]{Name \ref{label}}
```

instead of just:

```
Name \ref{label}
```

**Comment by Jellby** [comment]
Proposed fix for the second part (short links): in sphinx/writers/latex.py (1292), replace:

```
self.body.append(title.replace('#', ref))
```

with:

```
hyperref = '\\hyperref[%s]{%s}' % (self.idescape(id), title.replace('#', ref))
self.body.append(hyperref)
```

**Comment by tk0miya** [comment]
Fixed by PR #1752.
please try it. thanks.
